### PR TITLE
nc-news-7

### PR DESCRIPTION
**Pull request by mandeep-s1ngh**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -207,3 +207,108 @@ describe("app", () => {
...
+        .post("/api/articles/11/comments")
+        .send(requestBody)
+        .expect(400)
+        .then(({ body }) => {
+          ({ comment } = body);
+          expect(body.msg).toBe("Bad Request");
+        });
+    });
+  });
+});
```
Comment (corkesw) [inline __tests__/app.test.js:327]: We could do with a 404 test here for a non existent artilce_id

File: __tests__/app.test.js [hunk 2]
```diff
@@ -207,3 +207,108 @@ describe("app", () => {
...
+        .post("/api/articles/notAnId/comments")
+        .send(requestBody)
+        .expect(400)
+        .then(({ body }) => {
+          ({ comment } = body);
+          expect(body.msg).toBe("Bad Request");
+        });
+    });
+
+    it("400: POST - Responds with a 400 error msg for invalid schema validation", () => {
```
Comment (corkesw) [inline __tests__/app.test.js]: A test that would be useful here would be for a valid username that does not exists. We should test for a 404 in that case.

**Review by corkesw — COMMENTED** [review]
There are a couple of tests that I would like to see here - they will both generate psql errors. I'd be careful with passing a number as a username as I suspect it may be inserted as text into the db (or at least it will try to do that). You may actually be getting a foreign key violation (23503) which would want a 404 error as it means we have referred to a resource that does not exist in the linked table.

**Review by lewispricey — APPROVED** [review]
Great work on this Mandeep, thanks for implementing Stephen's feedback 👍 

The tests look really good and you're now covering every opportunity for the user to go wrong.

Good to merge when you're ready 🚀

**Pull request merged**